sql,jobs: add owner_id column to system.scheduled_jobs table

andyyang890 opened the pull request:
This patch adds a new `owner_id` column to the `system.scheduled_jobs`
table, which corresponds to the existing `owner` column, in order to
bring us closer to the eventual goal of allowing renaming of users.
Migrations are also added to alter and backfill the table in older
clusters.

Part of #87079

Release note: None

andyyang890 committed:
sql,jobs: add owner_id column to system.scheduled_jobs table

This patch adds a new `owner_id` column to the `system.scheduled_jobs`
table, which corresponds to the existing `owner` column, in order to
bring us closer to the eventual goal of allowing renaming of users.
Migrations are also added to alter and backfill the table in older
clusters.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/98737)
<!-- Reviewable:end -->

andyyang890 committed:
sql: add basic implementation of `func (p *planner) UserID()`

This patch adds a basic implementation of `func (p *planner) UserID()`
to help cut down on repeated code. Eventually, we'll want to cache the
user ID somewhere to prevent having to perform a lookup each time.

Release note: None

andyyang890 committed:
sql: add IsActive to PlanHookState

Release note: None

andyyang890 committed:
WIP compaction schedule gating

andyyang890 committed:
WIP schema telemetry schedule gating

andyyang890 committed:
WIP backup schedule gating

andyyang890 committed:
WIP changefeed schedule gating
